oceanwaves/spectral.py

```diff
@@ -45,17 +73,21 @@ def jonswap(f, Hm0, Tp, gamma=3.3, sigma_low=.07, sigma_high=.09,
     else:
         raise ValueError('Unknown method: %s' % method)
 
-    E_pm  = alpha * Hm0**2 * Tp**-4 * f**-5 * np.exp(-1.25 * (Tp * f)**-4)
-        
+    E_pm = alpha * Hm0**2 * Tp**-4 * f**-5 * np.exp(-1.25 * (Tp * f)**-4)
+
     # JONSWAP
     sigma = np.ones(f.shape) * sigma_low
     sigma[f > 1./Tp] = sigma_high
 
-    E_js = E_pm * gamma**np.exp(-0.5 * (Tp * f - 1)**2. / sigma**2.);
-    
+    E_js = E_pm * gamma**np.exp(-0.5 * (Tp * f - 1)**2. / sigma**2.)
+
     if normalize:
-        E_js *= Hm0**2. / (16. * trapz_and_repeat(E_js, f, axis=-1))
-        
+        if type(Hm0) == np.ndarray:
```

Comment:
I would say that `axis=0` works for both cases. Can you check that?
If not, it would be better to set the value for `axis` in the if-clause and then update `E_js`.
That way you don't store the normilization intelligence twice, which is prone to errors. So:

```
if isinstance(Hm0, np.ndarray):
    axis = 0
else:
    axis = -1
E_js *= Hm0**2. / ( ... trapz_and_repeat(..., axis=axis))
```

Reply:
axis=0 worked in all my tests, so I update the code to reflect it.